### DISCUSSION TITLE
Non-technical person here! Is it possible to save full websites with this?

### DISCUSSION THREAD

**Discussion by anadventcalendar** [gildas-lormeau/SingleFile#1496] [Q&A]
Specifically wondering if it is possible to save full websites in Safari (e.g., similar to Wayback Machine, where you can navigate to different pages of an old website). If SimpleFile cannot, are there other methods? I don't need to save it to my computer nessecarily, a public site like Wayback Machine is actually even better (saves me storage room and saves the website for others too!)

Web Archive/Wayback Machine seemingly only has a browser extension for Chrome, which I imagine is more convenient than logging in and waiting for the site to download/save the requested website content. Regardless, the Wayback Machine website won't even load for me in Safari currently (maybe it's down?). 

Thanks! Sorry for the probably stupid question.

**Comment by anadventcalendar** [comment]
also, Singlefile isn't showing up within my safari extensions. I've shutdown and restarted the computer, and Safari. Not sure what's causing it. When opening the SingleFile app, the button to quit and open Safari Extension Preferences does not do anything when clicking.

**Comment by gildas-lormeau** [comment]
SingleFile as an extension is not really suitable for saving complete sites. However, this functionality is possible with [SingleFile CLI](https://github.com/gildas-lormeau/single-file-cli) (see last examples in the README). From a technical point of view, Wayback Machine relies on the WARC/WACZ format for this purpose. I recommend that you take a look at the existing file format tools designed around this file format.

For the problem in Safari, do you see SingleFile in the Safari settings (cf. Extensions tab)?

**Comment by wanghaisheng** [comment]
@anadventcalendar https://github.com/webrecorder
